src/rpdk/core/submit.py

```diff
@@ -12,6 +12,10 @@
 def submit(args):
     project = Project()
     project.load()
+    # Update config file if use-docker or no-docker used in submit call
+    if args.use_docker or args.no_docker:
+        project.settings["use_docker"] = args.use_docker
```

Comment:
If I understand correctly, feeding this CLI arg into `project.settings` will write the new flags to `.rpdk-config.json` and get picked up by future CLI commands. This may not be desirable if the user only wants to override this setting for a single run. Perhaps a better solution would be to extend project settings with a flag or something that lets you control whether the setting gets persisted or not.

Also, I notice that only the python plugin uses the `use_docker` setting today. The typescript plugin uses `useDocker`, and as far as I can tell neither the java or go plugins support building in a docker container at all. I think it's worth considering whether this behaviour can be standardised across all plugins (but that could be a future enhancement / PR).

Reply:
Typescript was updated to match the other two plugins a little while back and I tested this update on python and typescript extensions (hooks/resources)  I don't have a golang extension handy but it should also apply.

Since config file actually only gets written during the "cfn init" portion of things.  The override won't update the .rpdk-config file.

Reply:
BTW I'm basing this off the current master branch of the typescript plugin.  The next release of the typescript plugin will have this use_docker update.

Reply:
Whoops just noticed that comment, let me update that to be accurate :)